### PR TITLE
`flake.nix` and ci with cachix

### PR DESCRIPTION
**Pull request by quinn-dougherty**
I needed to do this for a project, and I figured I'd suggest it upstream in case yall are interested.

You'll need a `CACHIX_AUTH_TOKEN` for the ci that I can email you to be set in the project settings (the `ci-cachix.yml` file will fail without it). adding OSX to the build matrix should work fine out of the box, but I didn't want to go ahead and do that unless it was expressed to be desired by the team, since the build minutes add up on osx last time I checked. 

lmk if you want to cherry pick aspects of the PR but not the whole thing!

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (cristianoc) [inline yarn.lock:1]: There's already a package lock in the same place.

Comment (quinn-dougherty) [reply yarn.lock:1]: I'm using the `yarn2nix` pattern, which requires `yarn.lock`. the `package-lock.json` version (called `node2nix`) has some more overhead and is less pure (requires unsandbox'd internet access)

**Comment by cristianoc** [comment]
Notice the project has moved inside the compiler (starting from the future 10.1 release) but happy to merge this in the meanwhile.

File: yarn.lock (hunk 1 above)
Comment (cristianoc) [reply yarn.lock:1]: Should I merge as-is then?

Comment (quinn-dougherty) [reply yarn.lock:1]: if you don't mind having both `yarn` and `npm` lockfiles, then it's ready.

Comment (cristianoc) [reply yarn.lock:1]: This repo is not going to be used for much longer, so no point being picky.

**Pull request merged**